### ISSUE TITLE
Adding depth or similar function to Pose preprocessor

### ISSUE THREAD

**Issue by Drakmour** [Mikubill/sd-webui-controlnet#51]
This tool is a blast! Especially the pose one! Absolutely brilliant. The only thing it lacks is somekind detect the depth (like the script does) of characters, determine which are closer, which are farther, and what limbs or body parts are behind the body or objects. For example if the model put her hands behind back, it thinks just that they are not visible and make non visible parts random but in front, not behind. And same for characters that are behind other characters, it tries to draw them not behind. I can't fully suggest how to achieve it but it would be awesome!

P.S. I tried to use both Controlnet + depth script, and it somewhy drags images from original image even when denoise is 1.

**Comment by Dekker3D** [comment]
This seems like another case where it'd be great to be able to combine different ControlNet inputs.

**Comment by Drakmour** [comment]
> This seems like another case where it'd be great to be able to combine different ControlNet inputs.

yeah, I saw your thread after posting mine and I saw your request is kinda similar to mine. :-) I hope they'll add it, it will so much freedom for executing the pose from head. :-)
